### DISCUSSION TITLE
[@types/office-js] WorkbookCreated not usable

### DISCUSSION THREAD

**Discussion by lbourda** [DefinitelyTyped/DefinitelyTyped#53921] [Issues with a @types package] [answered]
As per the documentation, "the WorkbookCreated object is the top level object created by Application.CreateWorkbook."

Two problems here:
1. the function createWorkbook is actually in Excel and not in Application
2. the function does not return a WorkbookCreated object but a void:
`	export function createWorkbook(base64?: string): Promise<void>;`

On the other hand, a similar object and function exist in Word:
"The DocumentCreated object is the top level object created by Application.CreateDocument."
`	createDocument(base64File?: string): Word.DocumentCreated;`

Additionally, compared to DocumentCreated, WorkbookCreated does not have a whole lot of properties/methods to use.

**Comment by ElizabethSamuel-MSFT** [comment]
@lbourda Apologies for our delayed response.

Tagging @alison-mk to take a look.

Thanks.

**Comment by alison-mk** [comment, marked as answer]
Hi @lbourda, thank you for submitting this issue. I've researched your question internally. Both the `Application.createWorkbook` and `Excel.createWorkbook` APIs exist. We recommend using [Excel.createWorkbook](https://learn.microsoft.com/javascript/api/excel?view=excel-js-preview#excel-excel-createworkbook-function(1)), which returns `Promise<void>` as you describe. 

See the [Create workbook](https://github.com/OfficeDev/office-js-snippets/blob/main/samples/excel/50-workbook/create-workbook.yaml) snippet for more information about how to work with `Excel.createWorkbook`.

Thanks, 
Alison
